tests/src/Integration/Engine/AnnotationProcessingTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Engine\AnnotationProcessingTest.php.
+ */
+
+namespace Drupal\Tests\rules\Integration\Engine;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+use Drupal\Core\Session\SessionManagerInterface;
+
+/**
+ * Test auto saving of variables after Rules execution.
+ *
+ * @group rules
+ */
+class AnnotationProcessingTest extends RulesIntegrationTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+    $this->enableModule('user');
+    $this->sessionManager = $this->prophesize(SessionManagerInterface::class);
+    $this->container->set('session_manager', $this->sessionManager->reveal());
+  }
+
+  /**
+   * Tests if our ContextDefiniton annotations are correctly processed.
+   *
+   * @dataProvider provideRulesPlugins
+   *
+   * @param array $params
+   *   An associative array containing type information from Rules plugins.
+   *
+   * @group rules
+   */
+  public function testCheckConfiguration($params) {
```

Comment:
instead of $params can we separate out all 4 parameters to make the test easier to read?